blueprint-test-utils/src/lib.rs

```diff
@@ -0,0 +1,214 @@
+        url: input.local_tangle_node,
+        bootnodes: input.bootnodes,
+        node_key: Some(node_key),
+        base_path: PathBuf::from(input.base_path),
+        keystore_password: None,
+        chain: SupportedChains::LocalTestnet,
+        verbose: input.verbose,
+        pretty: input.pretty,
+    };
+
+    let shutdown_signal = futures::future::pending();
+
+    blueprint_manager::run_blueprint_manager(
+        blueprint_manager_config,
+        gadget_config,
+        shutdown_signal,
+    )
+    .await
+    .expect("Failed to run blueprint manager")
+}
+
+/// Sets up the default logging as well as setting a panic hook for tests
+pub fn setup_log() {
+    let _ = SubscriberBuilder::default()
+        .with_env_filter(EnvFilter::from_default_env())
+        .finish()
+        .try_init();
+
+    std::panic::set_hook(Box::new(|info| {
+        log::error!(target: "gadget", "Panic occurred: {info:?}");
+        std::process::exit(1);
+    }));
+}
+
+pub async fn create_blueprint(
+    client: &TestClient,
+    account_id: &sr25519::Keypair,
+    blueprint: Blueprint,
+) -> Result<(), Box<dyn Error>> {
+    let call = api::tx().services().create_blueprint(blueprint);
+    let res = client
+        .tx()
+        .sign_and_submit_then_watch_default(&call, account_id)
+        .await?;
+    res.wait_for_finalized_success().await?;
+    Ok(())
+}
+
+pub async fn register_blueprint(
+    client: &TestClient,
+    account_id: &sr25519::Keypair,
+    blueprint_id: u64,
+    preferences: Preferences,
+    registration_args: RegistrationArgs,
+) -> Result<(), Box<dyn Error>> {
+    let call = api::tx()
+        .services()
+        .register(blueprint_id, preferences, registration_args);
+    let res = client
+        .tx()
+        .sign_and_submit_then_watch_default(&call, account_id)
+        .await?;
+    res.wait_for_finalized_success().await?;
+    Ok(())
+}
+
+pub async fn submit_job(
+    client: &TestClient,
+    user: &sr25519::Keypair,
+    service_id: u64,
+    job_type: Job,
+    job_params: Args,
+) -> Result<(), Box<dyn Error>> {
+    let call = api::tx().services().call(service_id, job_type, job_params);
+    let res = client
+        .tx()
+        .sign_and_submit_then_watch_default(&call, user)
+        .await?;
+    let _res = res.wait_for_finalized_success().await?;
+    Ok(())
+}
+
+pub async fn wait_for_completion_of_job(
+    client: &TestClient,
+    service_id: u64,
+    job_id: u64,
+) -> Result<JobResults, Box<dyn Error>> {
+    loop {
+        gadget_io::tokio::time::sleep(Duration::from_millis(100)).await;
+        // Most of what we need is here: api::tx().services() [..] .create_blueprint()
+        let call = api::storage().services().job_results(service_id, job_id);
+        let res = client.storage().at_latest().await?.fetch(&call).await?;
+        // client.tx() to send something to the chain
+
+        if let Some(ret) = res {
+            return Ok(ret);
+        }
+    }
```

Comment:
I would rather listen to events from the client and for each block I find the `JobResultSubmitted` event on which I will check for the call id and job_id.